### PR TITLE
`Iris` Enhancements of UX in the Chatbot Widget

### PR DESCRIPTION
**Pull request by undernagruzez**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [ ] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [ ] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [ ] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [ ] I documented the Java code using JavaDoc style.
#### Client
- [ ] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [ ] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [ ] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] I documented the TypeScript code using JSDoc style.
- [ ] I added multiple screenshots/screencasts of my UI changes.
- [ ] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Description
This PR addresses various UX problems.
1. Scroll to the bottom button was fixed. How it won't appear immediately and will let you scroll approximately through 1 message.
2. Empty message div disappears completely, instead of setting opacity to 0.
3. Pyris exceptions are now transmitted via Artemis to the client.
4. When disabling Iris during interactions with the chatbot, a corresponding error message will be shown.
5. Iris messages should be synchronized across multiple browser tabs.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to Course Administration Section
3. Enable Iris on all levels
4. Open a an exercise page on 2 tabs
5. Go into the exercise view and start chatting with the LLM
6. Ensure the messages appear on both tabs, without any duplications
7. Disable Iris on exercise level while leaving the chat widget open in another tab
8. Send something, ensure a correct error message appears
9. Reenable Iris
10. Send 5-6 messages, make sure scrolling gets enabled. Make sure the arrow appears (not immediately).
11. If you can mock Pyris responses(locally or on Pyris VM), do so, and check that correct error messages appear. Alternatively, navigate to global Iris settings and modify the template to an invalid one.

#### Performance Review
- [ ] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisMessageResource.java [hunk 1]
```diff
@@ -76,6 +82,10 @@ public ResponseEntity<IrisMessage> createMessage(@PathVariable Long sessionId, @
         irisSessionService.checkHasAccessToIrisSession(session, null);
         var savedMessage = irisMessageService.saveMessage(message, session, IrisMessageSender.USER);
         irisSessionService.requestMessageFromIris(session);
+        savedMessage.setNonce(message.getNonce());
+        CompletableFuture.delayedExecutor(500, TimeUnit.MILLISECONDS).execute(() -> {
```
Comment (Hialus) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisMessageResource.java:86]: Why do you delay this? This just adds unnecessary latency to the requests.

File: src/main/java/de/tum/in/www1/artemis/domain/iris/IrisMessage.java [hunk 2]
```diff
@@ -43,6 +44,9 @@ public class IrisMessage extends DomainObject {
     @Enumerated(EnumType.STRING)
     private IrisMessageSender sender;
 
+    @Transient
+    private Integer nonce;
```
Comment (Hialus) [inline src/main/java/de/tum/in/www1/artemis/domain/iris/IrisMessage.java:48]: What is this int used for? 
Also please provide a more descriptive name, as "number used only once" is not very helpful.

**Review by Hialus — CHANGES_REQUESTED** [review]
Left two first comments on the server changes, but I can't do more, as it contains changes from an unrelated PR.
Also please remove the Iris logo from this PR. It is not fully implemented and will be added in a separate PR.
Also from your PR description:
> 3. Pyris exceptions are now transmitted via Artemis to the client.
>4. When disabling Iris during interactions with the chatbot, a corresponding error message will be shown.
>5. Iris messages should be synchronized across multiple browser tabs.

3 is not related to this PR, but to PR #6819.
5 is this not already the case? The server should send all messages of a session through the web socket to all browser tabs.

**Comment by undernagruzez** [comment]
> 3 is not related to this PR, but to PR #6819.
> 5 is this not already the case? The server should send all messages of a session through the web socket to all browser tabs.

No, it is not possible to show error messages from Pyris in PR#6819. This PR builds on top of it.
Also, this was not already the case that the server was sending messages of a session through the web socket channel.

File: src/main/java/de/tum/in/www1/artemis/domain/iris/IrisMessage.java (hunk 2 above)
Comment (undernagruzez) [reply src/main/java/de/tum/in/www1/artemis/domain/iris/IrisMessage.java:48]: This variable is used to help the client to differentiate between messages send from the current tab and all other tabs.

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisMessageResource.java (hunk 1 above)
Comment (undernagruzez) [reply src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisMessageResource.java:86]: Was introduced in the course of development, and will be removed. However, this only adds latency to other tabs, not the tab that has just sent the message.

**Review by SabrinaGlatz — APPROVED** [review]
tested on ts9, works as expected.

**Pull request merged**